tests/baselines/reference/functionReadonlyLength.types

```diff
@@ -0,0 +1,45 @@
+=== tests/cases/compiler/functionReadonlyLength.ts ===
+const f0 = new Function()
+>f0 : Function
+>new Function() : Function
+>Function : FunctionConstructor
+
+f0.length = 1;
+>f0.length = 1 : 1
+>f0.length : any

```

Comment:
this feels wrong - perhaps the `.types` printer tries to only lookup the "target" symbol of this assignment and doesn't try to look into `Function` here etc. Should I try to fix this as part of this PR?

Reply:
This may be fine. What the type writer is saying is that the type of the _expression_ `f0.length = 1` is `1`, which is correct:

```js
// note: NOT in strict mode. In strict mode, attempting to assign to length will throw
function f(a, b) { }
console.log(f.length); // 2

const i = f.length = 1;
console.log(i); // 1
console.log(f.length); // 2
```

Reply:
I was specifically referring to the 9th line that has `f0.length : any` and it seems that this `any` is a little bit off here as I would expect this to be of type `number` 🤔

Reply:
Ah, but actually this is already the case for `new Function` case so perhaps this is how it's printed for the readonly LHS. It's still a little bit confusing to me - but at least my PR doesn't really change anything in this regard 🤷‍♂️